osgtest/library/files.py

```diff
@@ -144,12 +144,9 @@ def replace(path, old_line, new_line, owner=None, backup=True):
     write(path, lines_to_write, owner, backup)
 
 def replace_regexpr(path, regexp, new_line, owner=None, backup=True):
-    """Replace an old line that matches regexpr with a new line in given path.                                                     
-    The 'owner' and 'backup' arguments are passed to write().                                                                                  
+    """Replace an old line that matches regexpr with a new line in given path.
+    The 'owner' and 'backup' arguments are passed to write().
     """
-    match = re.search(regexp, new_line)
-    if not match:
-       raise ValueError("The new_line '%s' does not match reg expr '%s'" % (new_line, regexp)) 
     lines_to_write = []
```

Comment:
Hm, I am not sure that I agree with this change. At very least, I would have made the match optional and, via an argument, made it required by default. I think that the point was that one is *expecting* the match to work and hence the change to take place.

Another possibility is to have the function return the number of changed lines, such that the caller could examine the return and decide if 0 (no lines changed) is acceptable — but of course both the PR change and this idea change the semantics of this function and thus require going through test code to see if that is OK.